src/appleseed/renderer/modeling/project/projectfileupdater.cpp

```diff
@@ -2117,6 +2120,43 @@ namespace
                 }
             }
         }
+
+        static void update_assemblies(const AssemblyContainer& assemblies)
+        {
+            for (const auto& assembly: assemblies)
+            {
+                update_bsdfs(assembly.bsdfs());
+                update_assemblies(assembly.assemblies());
+            }
+        }
+
+        static void update_bsdfs(BSDFContainer& bsdfs)
+        {
+            for (auto& bsdf: bsdfs)
+            {
+                if (strcmp(bsdf.get_model(), "glass_bsdf") == 0)
+                    update_microfacet_params(bsdf);
+            }
+        }
+
+        static void update_microfacet_params(BSDF& bsdf)
+        {
+            ParamArray& params = bsdf.get_parameters();
+
+            const string mdf = params.get_optional<string>("mdf", "");
+            params.strings().remove("mdf");
+
+            if (mdf != "ggx")
```

Comment:
If an `mdf` param was not found, `mdf` is empty here and we will (incorrectly) print the message. Better:

    if (params.strings().exist("mdf"))
    {
        const string mdf = params.strings().get<string>("mdf");
        params.strings().remove("mdf");
        if (mdf != "ggx")
            ...
    }